### DISCUSSION TITLE
[CSS] Specific Modal for AQS Modals → Custom AQS Modals

### DISCUSSION THREAD

**Discussion by FelipeRearden** [tadashi-aikawa/obsidian-another-quick-switcher#129] [Ideas]
Hello @tadashi-aikawa !!!!

This is an idea that I would like to share with you about the possibility to let the users apply CSS snippets to change the size of AQS Modals 

AQS Versions v6.x and v7.x has such amazing features that are changing the core idea of AQS as QuickSwitcher to an Operating System AQSOS. 

Everything navigation that I do inside Obsidian in from a AQS Modal. I really mean everything: tags, yaml keys, headers backlinks and filenames. 

This way, I am start feel that I need more **space** to see all of this content that are inside an AQS Modal. 

I start to dig into console to see the CSS and I found that AQS modals use the standard  Obsidian Modals. 

📸 `.prompt {	max-width: 98vw; width: 100%; }` → Real Word exemple
![D97917EE-C64E-4FCA-AE25-279CA7DE4729](https://user-images.githubusercontent.com/85180696/203571407-89f68902-5da5-49e5-9cf0-c425bc694d84.png)

📸  Standard AQS width → Real Word exemple
![018A3944-D822-402B-A4EF-399E5CDF6D28](https://user-images.githubusercontent.com/85180696/203571565-2ddc9a30-a5e2-4f05-9dbd-308b9fd64275.png)

ℹ️ **If I change the** `.prompt` **it will hurt all Obsidian Modals**.

---

The idea that I want to share if wouldn't be a good idea to have specific modals for every AQS Modals ?

This way, we could customize the AQS Modals based on the type of information that we want  form each search command. 

---

Let me know what you think about this idea :)

Have a great day ☀️

**Comment by FelipeRearden** [comment]
Hello @tadashi-aikawa !!!

Sorry for bothering you 😀 

I am curious to know your opinion about this idea for AQS modals 🙃 

Have a great day!!!!

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 

> This way, I am start feel that I need more space to see all of this content that are inside an AQS Modal.

I think the best size depends on who uses AQS and how to use it. Therefore I want to respect the size of Obsidian's quick switcher as a default.

**Reply by FelipeRearden** [reply]
> I think the best size depends on who uses AQS and how to use it. Therefore I want to respect the size of Obsidian's quick switcher as a default.

100% understood @tadashi-aikawa !!!

Thanks for the reply!

I wish you a fantastic day!

**Comment by FelipeRearden** [comment]
@tadashi-aikawa 

ℹ️ Just one comment in case you review this discussion in the future. 

I 100% agree with you that the default position should respect Obsidian default Modals.  

→ My idea is about giving the option to advanced users that are comfortable with css to create css snippets to change the size of AQS Modals. It's a hidden feature only available to users that know what they are doing with the modals. 

I wish you a fantastic day ☀️

**Comment by FelipeRearden** [comment]
### 📰 Update 

Since the new Obsidian version supports `:has` selector. I was able to achieve this with this Snippet ...

```
.prompt:has(.another-quick-switcher__status__search-command) {
	max-width: 100vw; width: 100%;
}
```

I am pasting here in case someone else search for this in the future ☀️
